test/integration/orderFlowFee.behavior.t.sol

```diff
@@ -107,57 +115,275 @@ contract OrderFlowFeeTest is Test, ConsolidatedEvents {
+
+        // Account margin is emptied
+        assertEq(account.freeMargin(), 0);
+
+        // Market margin is reduced by (imposedOrderFlowFee - 10 000)
+        assertEq(
+            uint256(position.margin),
+            uint256(marginDelta) - (imposedOrderFlowFee - 10_000) - 563
+        );
+
+        // Assert that fee was correctly sent to treasury address
+        assertEq(treasuryPostBalance - treasuryPreBalance, imposedOrderFlowFee);
     }
 
-    /// @custom:todo desired fill price behaviour? floor/ceiling price behaviour? reverts? etc
-    /// @custom:todo what happens if withdrawing margin from market results in leverage exceeding allowed leverage? revert?
-    /// @custom:todo think deeply about the edge cases
+    /// Verifies that transaction reverts if there is insufficient margin to cover for orderFlowFee without exceeding positions limits
+    /// @dev Synthetix makes transaction reverts if the resulting position is too large, outside the max leverage, or is liquidating.
+    function test_imposeOrderFlowFee_market_margin_failed() public {
+        // Set orderflow fee high to easily test that transaction fails if neither account or market has sufficient margin to cover for fees
+        uint256 testOrderFlowFee = 10_000; // 10%
+        settings.setOrderFlowFee(testOrderFlowFee);
+
+        fundAccount(1000 ether);
+
+        uint256 treasuryPreBalance = sUSD.balanceOf(settings.TREASURY());
+
+        // create a long position in the ETH market
+        address market = getMarketAddressFromKey(sETHPERP);
+
+        /// Deposit all available account margin into market margin
+        int256 marginDelta = int256(1000 ether);
+        int256 sizeDelta = 1 ether;
+
+        (uint256 desiredFillPrice,) =
+            IPerpsV2MarketConsolidated(market).assetPrice();
+
+        // Deposit market margin
+        IAccount.Command[] memory commands = new IAccount.Command[](1);
+        commands[0] = IAccount.Command.PERPS_V2_MODIFY_MARGIN;
+        bytes[] memory inputs = new bytes[](1);
+        inputs[0] = abi.encode(market, marginDelta);
+        account.execute(commands, inputs);
+
+        // Execute Atomic Order
+        IAccount.Command[] memory commandsAtomic = new IAccount.Command[](2);
+        commandsAtomic[0] = IAccount.Command.PERPS_V2_SUBMIT_ATOMIC_ORDER;
+        bytes[] memory inputsAtomic = new bytes[](2);
+        inputsAtomic[0] = abi.encode(market, sizeDelta, desiredFillPrice);
+
+        // Current configuration should have insufficient margin to cover for orderflow fee and revert
+        // because there is not enough margin for the position delta.
+        vm.expectRevert("Insufficient margin");
 
-    /// @custom:todo use atomic order flow to simplify testing
+        account.execute(commandsAtomic, inputsAtomic);
+
+        uint256 treasuryPostBalance = sUSD.balanceOf(settings.TREASURY());
+
+        // Assert that no overflowfee was distributed
+        assertEq(treasuryPreBalance, treasuryPostBalance);
```

Comment:
hmm, maybe I am mistaken, but if a revert is caught and asserted, will the following code be reached??